simapp/helpers/test_helpers.go

```diff
@@ -18,31 +20,58 @@ const (
 )
 
 // GenTx generates a signed mock transaction.
-func GenTx(msgs []sdk.Msg, feeAmt sdk.Coins, gas uint64, chainID string, accnums []uint64, seq []uint64, priv ...crypto.PrivKey) authtypes.StdTx {
-	fee := authtypes.StdFee{ //nolint:staticcheck // SA1019: authtypes.StdFee is deprecated
-		Amount: feeAmt,
-		Gas:    gas,
-	}
-
-	sigs := make([]authtypes.StdSignature, len(priv)) //nolint:staticcheck // SA1019: authtypes.StdSignature is deprecated
+func GenTx(gen client.TxGenerator, msgs []sdk.Msg, feeAmt sdk.Coins, gas uint64, chainID string, accnums []uint64, seq []uint64, priv ...crypto.PrivKey) sdk.Tx {
```

Comment:
Yikes, I don't know how much lift this will require, but if it's not too much, ideally this should return an error instead of doing panics.

Reply:
I like the idea, will check if it's changing too much. Else it should be find to eliminate panics and return error